cdc/sink/mq/manager/kafka_manager.go

```diff
@@ -65,7 +75,16 @@ func (m *kafkaTopicManager) GetPartitionNum(topic string) (int32, error) {
 		return partitions.(int32), nil
 	}
 
-	return m.CreateTopic(topic)
+	partitionNum, err := m.CreateTopic(topic)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	err = m.WaitUntilTopicCreationDone(topic)
```

Comment:
Wouldn't it make more sense for it to be wrapped in a CreateTopic? Rather than exposing it to the caller?

Reply:
It is initially designed to be wrapped in `CreateTopic()`, but it is hard to do the topic creation delay mocking because the mocking should happen between **creating** the topic and **waiting** for the topic to be ready. So eventually it is exposed to the caller. Any suggestion to do the mocking?

Reply:
BTW, the topic creation delay mocking is in https://github.com/pingcap/tiflow/pull/5691/files#diff-caac373783c231dc1838768a3ec3260dc2b40a94946a2cc8b8cbc8e8bd7d118cR128

Reply:
add a method such  as CreateTopicWithRetry XXX , and invoke this method here?

Reply:
> add a method such as CreateTopicWithRetry XXX , and invoke this method here?

Done in the latest commit.